### DISCUSSION TITLE
DateTimePicker body defaulting to previous month

### DISCUSSION THREAD

**Discussion by eramosr16** [dotnetcore/BootstrapBlazor#4268] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

DateTimePicker.razor component defaults to previous month of whatever date is passed as value.

### Expected Behavior

Should default to current month passed as value

### Steps To Reproduce

Here is the reference on the code: https://github.com/dotnetcore/BootstrapBlazor/blob/7d9a6a9fefefac9b992a4061e71d3a7496362fc8/src/BootstrapBlazor/Components/DateTimePicker/DatePickerBody.razor.cs#L137

**Comment by bb-auto** [comment]
@eramosr16 Thank you for filing this issue. In order for us to investigate this issue, please provide [a minimalistic repro project](https://github.com/dotnetcore/bootstrapblazor/blob/main/docs/repro.md) that illustrates the problem.

**Comment by ArgoZhang** [comment]
@eramosr16 hi this is by design.

**Reply by eramosr16** [reply]
Sorry can you elaborate a little more on why is by design, is there a logical explanation on why it shows the previous month as default instead of the current month date. 
If I'm setting the initial date of the component I'm expecting to see the current month of the initial date selected not the previous month.

**Reply by eramosr16** [reply]
@ArgoZhang Any update on this?

**Reply by ArgoZhang** [reply]
@eramosr16 sorry for the delayed reply. this component was designed by `Element UI`

https://element.eleme.cn/#/en-US/component/date-picker

so. If you change it to allow you to select any time, you may need to change more code to implement this function.

**Reply by eramosr16** [reply]
Hi @ArgoZhang I basically don't want to do any additional changes to the component just basically remove the -1 I attached on the screenshot in the ticket to display the current month instead of the previous one.

**Reply by eramosr16** [reply]
Any updates here?.

**Reply by ArgoZhang** [reply]
Can you submit a PR to meet your needs? I don't want to change this feature at the moment, because that would affect the functionality of many old users. Or we can add a parameter to control whether to always display the previous month

**Reply by eramosr16** [reply]
@ArgoZhang After digging further I noticed the the bug is not related to the component logic, is related to the Single View Mode that takes the left panel and this panel is the previous month the pr created here is to address this issue and make the single view more consistent: https://github.com/dotnetcore/BootstrapBlazor/pull/4357
Apologies for the confusion before, since I didn't explain myself better regarding this issue.

**Reply by ArgoZhang** [reply]
@eramosr16 well done! later I will take a review. thank you.
